ts/components/messages/MessagesArchive.tsx

```diff
@@ -143,10 +150,20 @@ class MessagesArchive extends React.PureComponent<Props, State> {
     this.state = {
       lastMessagesState: pot.none,
       filteredMessageStates: [],
-      allMessageIdsState: none
+      allMessageIdsState: none,
+      isErrorLoading: false
     };
   }
 
+  public componentDidUpdate(_: Props, prevState: State) {
+    this.setState({
```

Comment:
I'm quite sure you don't need to implement _componentDidUpdate_
It's enough to check if `this.props.messagesState` is error.
So try to define a const in render function
`const isErrorLoading = pot.isError(this.props.messagesState);`
and use it instead of state